src/avatar/Avatar.js

```diff
@@ -78,7 +78,7 @@ const AvatarComponent = ({
     ));
 
   // Remove placeholder styling if we're not using image
-  const hidePlaceholder = !source;
+  const hidePlaceholder = !source || !source.uri;
```

Comment:
I think using `&& ` instead of `||` would be a bit optimized as the engine will check the first argument and if it's false then it won't check for the second one (right of the operator).

Reply:
Noted, but if `source` doesn't contain `uri`, the app crashes because it can't found the property `uri` to compare. That's why I set the `||` instead of `&&`. Let me know what you think.

Reply:
Hi Jehiel, as mentioned in my previous comment that all JS engines come with a small hack/shortcut for optimizing such expressions.

```js
const x = null;

if (x && x.y) {
  console.log("x is !null and x.y as well");
} else {
  console.log("x is empty");
}

```

Just run the above snippet in Browser or Node.js and it won't fail because `x.y` will never be checked.

Reply:
I recommend that you test the following changes that this PR brings locally as well. You can use this app -> https://github.com/react-native-elements/react-native-elements-app
and link the local project using `npm link`. :)

Reply:
@pranshuchittora this is what I'm getting when changing the `||` to `&&` when testing
```
Avatar Component › Placeholders › shouldn't show placeholder if not using source

    TypeError: Cannot read property 'uri' of undefined

      79 | 
      80 |   // Remove placeholder styling if we're not using image
    > 81 |   const hidePlaceholder = !source && !source.uri;
```

Any help how to fix this?

Reply:
Hi so as mentioned this works for expressions , so instead try this.

`  const hidePlaceholder = (source && source.uri) ;`